components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -208,6 +217,199 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
+     * @return OAuth2AuthorizeRespDTO
+     */
+    public OAuth2AuthorizeRespDTO handleAuthorization(OAuthAuthzReqMessageContext authzReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        OAuth2AuthorizeReqDTO authzReqDTO = authzReqMsgCtx.getAuthorizationReqDTO();
+        ResponseTypeHandler authzHandler = getResponseHandler(authzReqDTO);
+
+        OAuth2AuthorizeRespDTO authorizeRespDTO = new OAuth2AuthorizeRespDTO();
+        if (isInvalidClient(authzReqDTO, authorizeRespDTO, authzReqMsgCtx, authzHandler)) {
+            return authorizeRespDTO;
+        }
+        if (isInvalidAccessDelegation(authzReqDTO, authorizeRespDTO, authzReqMsgCtx, authzHandler)) {
+            return authorizeRespDTO;
+        }
+        if (isErrorResponseFound(authorizeRespDTO)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error response received for authorization request by user : " + authzReqDTO.getUser() +
+                        ", client : " + authzReqDTO.getConsumerKey() + ", scopes: " +
+                        OAuth2Util.buildScopeString(authzReqDTO.getScopes()));
+            }
+            return authorizeRespDTO;
+        }
+        try {
+            // set the authorization request context to be used by downstream handlers. This is introduced as a fix for
+            // IDENTITY-4111
+            OAuth2Util.setAuthzRequestContext(authzReqMsgCtx);
+            authorizeRespDTO = authzHandler.issue(authzReqMsgCtx);
+        } finally {
+            // clears authorization request context
+            OAuth2Util.clearAuthzRequestContext();
+        }
+        return authorizeRespDTO;
+    }
+
+
+    /**
+     * validated requested scopes.
+     *
+     * @param authzReqMsgCtx OAuthAuthzReqMessageContext
+     * @param authzHandler   ResponseTypeHandler
+     */
+    private void validateRequestedScopes(OAuthAuthzReqMessageContext authzReqMsgCtx,
+                                         ResponseTypeHandler authzHandler) throws IdentityOAuth2Exception,
+            IdentityOAuth2UnauthorizedScopeException {
+
+        String[] requestedOIDCScopes;
+        try {
+            // Get OIDC scopes from requested scopes. At end of the scope validation OIDC scope will add to the
+            // approved scope list.
+            requestedOIDCScopes = OAuth2Util.getRequestedOIDCScopes(authzReqMsgCtx.getAuthorizationReqDTO()
+                    .getScopes());
+            // OIDC scopes are not validated in the scope validation process. Hence, Removing OIDC scopes from the
+            // requested scopes by the app.
+            String[] oidcRemovedScopes = OAuth2Util.removeOIDCScopesFromRequestedScopes(authzReqMsgCtx
+                    .getAuthorizationReqDTO().getScopes());
+            authzReqMsgCtx.getAuthorizationReqDTO().setScopes(oidcRemovedScopes);
+        } catch (IdentityOAuthAdminException e) {
+            throw new IdentityOAuth2Exception(e.getMessage());
```

Comment:
Can we have a proper message rather than sending the message from e.GetMessage()

Reply:
fixed